src/components/Modal/index.tsx

```diff
@@ -1,116 +1,118 @@
-import { cn } from "@/utils";
-import { ModalBodyStyles, ModalOverlayStyles } from "./index.style";
-import { ModalContextProvider } from "../../context/Modal/index";
-
-import { IoMdClose } from "react-icons/io";
-import { Button } from "../Button";
-
-import { ComponentProps, useEffect, useState } from "react";
-import { VariantProps } from "class-variance-authority";
-import { useModalContext } from "@/hooks";
+/** @jsxImportSource @emotion/react */
 import { createPortal } from "react-dom";
+import {
+  modalBodyBaseStyles,
+  modalContentStyles,
+  modalHeaderStyles,
+  modalOverlayStyles,
+} from "./index.style";
+import { ComponentProps, useContext } from "react";
+import { CSSObject, SerializedStyles } from "@emotion/react";
+import { Button } from "..";
+import { IoMdClose } from "react-icons/io";
+import { ModalContext, ModalContextProvider } from "@/context";
 
-export type ModalDivProps = ComponentProps<"div">;
-
-export type ModalProps = ModalDivProps &
-  VariantProps<typeof ModalBodyStyles> & {
-    isOpen: boolean;
-    onClose: () => void;
-    closeOnOutsideClick?: boolean;
-    scrollBehaviour?: boolean;
-    showCross?: boolean;
-    size?: "sm" | "md" | "lg" | "full";
-  };
-
-export type ModalTriggerProps = ModalDivProps & {
-  isOpen: boolean;
-  onClose: () => void;
+type ModalHeader = ComponentProps<"div"> & {
+  children: React.ReactNode;
+  className?: string;
+  css?: SerializedStyles | CSSObject;
 };
 
-export const Portal: React.FC<{ children: React.ReactNode }> = ({ children }) => {
-  const [mounted, setMounted] = useState(false);
+type ModalContent = ComponentProps<"div"> & {
+  children: React.ReactNode;
+  className?: string;
+  css?: SerializedStyles | CSSObject;
+};
 
-  useEffect(() => {
-    setMounted(true);
-    return () => setMounted(false);
-  }, []);
+type ModalFooter = ComponentProps<"div"> & {
+  children: React.ReactNode;
+  className?: string;
+  css?: SerializedStyles | CSSObject;
+};
 
-  return mounted
-    ? createPortal(children, document.getElementById("portal-root") || document.body)
-    : null;
+type ModalProps = ComponentProps<"div"> & {
+  show: boolean;
+  size?: "sm" | "md" | "lg" | "full";
+  children: React.ReactNode;
+  showCloseButton?: boolean;
+  closeOnOutsideClick?: boolean;
+  onCloseButtonClick: () => void;
```

Comment:
Suggestion: Prop Naming Improvements

- Use closable instead of showCloseButton
- Use maskClosable instead of closeOnOutsideClick
- Use onClose instead of onCloseButtonClick